targets/playground/examples/thymio-default-behaviours.aesl

```diff
@@ -341,15 +344,20 @@ if run_behavior == START then
 			sound_done = 0
 		end
 		callsub tableborder
-		
+
 	elseif state == EXPLORER then
 		# Breitenberg obtacle avoidance
 		if speed > 0 then
-			temp = (prox.horizontal[0]*1+prox.horizontal[1]*2+prox.horizontal[2]*3+prox.horizontal[3]*2+prox.horizontal[4]*1)
-			temp2 = prox.horizontal[0]*-4+prox.horizontal[1]*-3+prox.horizontal[3]*3+prox.horizontal[4]*4
-			motor.left.target = speed - (temp+temp2) / (2000/speed)
-			motor.right.target = speed - (temp-temp2) / (2000/speed)
+			call math.dot(temp,  avoidanceKernel1[0:4], prox.horizontal[0:4], 1)
+			call math.dot(temp2, avoidanceKernel2[0:4], prox.horizontal[0:4], 1)
 			callsub motorbound
+
+			call math.muldiv(temp, temp, 2000, speed)
+			call math.muldiv(temp2, temp2, 2000, speed)
+
+			motor.left.target  = (speed - temp) - temp2
```

Comment:
It is more natural wrt robotics and symmetric with the next line to write `speed - (temp + temp2)`.

Reply:
Good call.